Graphql checkout tutorial (#4271)
magento/devdocs: guides/v2.3/graphql/tutorials/checkout/checkout-shipping-method.md

---
layout: tutorial
group: graphql
title: Step 6. Set the shipping method
subtitle: GraphQL checkout tutorial
level3_subgroup: graphql-checkout
return_to:
  title: GraphQL Overview
  url: graphql/index.html
menu_order: 60
functional_areas:
  - Integration
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

The `setShippingMethodsOnCart` mutation defines the shipping methods for your order. It requires these input parameters:
 - `cart_id`
 - `carrier_code`
 - `method_code`

> {{ CART_ID }}` is the unique shopping cart ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html).

Author: **fixed**

---
layout: tutorial
group: graphql

> title: Step 6. Set shipping method

Reviewer: ```suggestion
title: Step 6. Set the shipping method
```

Author: **done**

subtitle: GraphQL checkout tutorial
level3_subgroup: graphql-checkout
return_to:
  title: GraphQL Overview
  url: graphql/index.html
menu_order: 60
functional_areas:
  - Integration
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

> `setShippingMethodsOnCart` mutation query is using to define a shipping method for your order and requires the next input parameters:

Reviewer: ```suggestion
The `setShippingMethodsOnCart` mutation defines the shipping methods for your order. It requires these input parameters:
```

Author: **done**

 - `cart_id`
 - `carrier_code`
 - `method_code`

> 

Reviewer: Add this below:

{{ CART_ID }}` is the unique shopping cart ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html).

Author: **done**

**Request**

The following mutation query assigns UPS "Ground" method.

```text
mutation {
  setShippingMethodsOnCart(input: {
    cart_id: "{{ CART_ID }}"
    shipping_methods: [
      {
        carrier_code: "ups"
        method_code: "GND"
      }
    ]
  }) {
    cart {
      shipping_addresses {
        selected_shipping_method {
          carrier_code
          method_code
          label
        }
      }
    }
  }
}
```

> where

Reviewer: Delete these lines

Author: **done**

`{{ CART_ID }}` - shopping cart unique ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html).

**Response**

```json
{
  "data": {
    "setShippingMethodsOnCart": {
      "cart": {
        "shipping_addresses": [
          {
            "selected_shipping_method": {
              "carrier_code": "ups",
              "method_code": "GND",
              "label": "United Parcel Service - Ground"
            }
          }
        ]
      }
    }
  }
}
```

{:.bs-callout .bs-callout-info}

> Send customer's authorization token in the `Authorization` parameter of the header if you set shipping method for a registered customer. See ["Get customer authorization token"]({{ page.baseurl }}/graphql/get-customer-authorization-token.html) to get more details.

Reviewer: Use the same paragraph updated elsewhere

Author: **done**

---
layout: tutorial
group: graphql
title: Step 6. Set the shipping method
subtitle: GraphQL checkout tutorial
level3_subgroup: graphql-checkout
return_to:
  title: GraphQL Overview
  url: graphql/index.html
menu_order: 60
functional_areas:
  - Integration
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

> The `setShippingMethodsOnCart` mutation defines the shipping methods for your order. It requires these input parameters:

Author: **done**

**Request**

The following mutation query assigns UPS "Ground" method.

```text
mutation {
  setShippingMethodsOnCart(input: {
    cart_id: "{{ CART_ID }}"
    shipping_methods: [
      {
        carrier_code: "ups"
        method_code: "GND"
      }
    ]
  }) {
    cart {
      shipping_addresses {
        selected_shipping_method {
          carrier_code
          method_code
          label
        }
      }
    }
  }
}
```

**Response**

```json
{
  "data": {
    "setShippingMethodsOnCart": {
      "cart": {
        "shipping_addresses": [
          {
            "selected_shipping_method": {
              "carrier_code": "ups",
              "method_code": "GND",
              "label": "United Parcel Service - Ground"
            }
          }
        ]
      }
    }
  }
}
```

{:.bs-callout .bs-callout-info}
For logged-in customers, send the customer's authorization token in the `Authorization` parameter of the header. See ["Get customer authorization token"]({{ page.baseurl }}/graphql/get-customer-authorization-token.html) for more information.

> ### Verify this step {#verify-step}

Author: **done**

1. Sign in as a customer to the website using the email `john.doe@example.com` and password `b1b2b3l@w+`.

2. Go to Checkout. 

> 3. Selected shipping method is displayed in the Shipping Methods section on the Shipping step.

Author: **done**

---
layout: tutorial
group: graphql
title: Step 6. Set the shipping method
subtitle: GraphQL checkout tutorial
level3_subgroup: graphql-checkout
return_to:
  title: GraphQL Overview
  url: graphql/index.html
menu_order: 60
functional_areas:
  - Integration
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

The `setShippingMethodsOnCart` mutation defines the shipping methods for your order. It requires these input parameters:
 
 * `cart_id`
 * `carrier_code`
 * `method_code`

`{ CART_ID }` is the unique shopping cart ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html).

**Request**

The following mutation query assigns UPS "Ground" method.

> 

Reviewer: Add the same note from step 4 here.

Author: **done**

```text
mutation {
  setShippingMethodsOnCart(input: {
    cart_id: "{ CART_ID }"
    shipping_methods: [
      {
        carrier_code: "ups"
        method_code: "GND"
      }
    ]
  }) {
    cart {
      shipping_addresses {
        selected_shipping_method {
          carrier_code
          method_code
          label
        }
      }
    }
  }
}
```

**Response**

```json
{
  "data": {
    "setShippingMethodsOnCart": {
      "cart": {
        "shipping_addresses": [
          {
            "selected_shipping_method": {
              "carrier_code": "ups",
              "method_code": "GND",
              "label": "United Parcel Service - Ground"
            }
          }
        ]
      }
    }
  }
}
```

> {:.bs-callout .bs-callout-info}

Author: **done**
